src/EncompassRest/ApiObject.cs

```diff
@@ -64,21 +64,21 @@ internal ApiObject(EncompassRestClient client, string baseApiPath)
 
-        internal Task PatchPopulateDirtyAsync(string requestUri, HttpContent content, string methodName, string resourceId, CancellationToken cancellationToken, IDirty target, bool populate) => PopulateDirtyInternalAsync(s_patchMethod, requestUri, content, methodName, resourceId, cancellationToken, target, populate);
+        internal Task PatchPopulateDirtyAsync(string requestUri, string queryString, HttpContent content, string methodName, string resourceId, CancellationToken cancellationToken, IDirty target, bool populate) => PopulateDirtyInternalAsync(s_patchMethod, requestUri, queryString, content, methodName, resourceId, cancellationToken, target, populate);
 
         internal Task<T> PatchAsync<T>(string requestUri, string queryString, HttpContent content, string methodName, string resourceId, CancellationToken cancellationToken, Func<HttpResponseMessage, Task<T>> func) => SendAsync(s_patchMethod, requestUri, queryString, content, methodName, resourceId, cancellationToken, func);
 
         internal Task PutAsync(string requestUri, string queryString, HttpContent content, string methodName, string resourceId, CancellationToken cancellationToken) => SendAsync<string>(HttpMethod.Put, requestUri, queryString, content, methodName, resourceId, cancellationToken, null);
 
         internal Task<string> PutRawAsync(string requestUri, string queryString, HttpContent content, string methodName, string resourceId, CancellationToken cancellationToken) => SendAsync(HttpMethod.Put, requestUri, queryString, content, methodName, resourceId, cancellationToken, ReadAsStringFunc);
 
-        internal Task PutPopulateDirtyAsync(string requestUri, HttpContent content, string methodName, string resourceId, CancellationToken cancellationToken, IDirty target, bool populate) => PopulateDirtyInternalAsync(HttpMethod.Put, requestUri, content, methodName, resourceId, cancellationToken, target, populate);
+        internal Task PutPopulateDirtyAsync(string requestUri, string queryString, HttpContent content, string methodName, string resourceId, CancellationToken cancellationToken, IDirty target, bool populate) => PopulateDirtyInternalAsync(HttpMethod.Put, requestUri, queryString, content, methodName, resourceId, cancellationToken, target, populate);
 
         internal Task<T> PutAsync<T>(string requestUri, string queryString, HttpContent content, string methodName, string resourceId, CancellationToken cancellationToken, Func<HttpResponseMessage, Task<T>> func) => SendAsync(HttpMethod.Put, requestUri, queryString, content, methodName, resourceId, cancellationToken, func);
 
         internal Task<bool> DeleteAsync(string requestUri, CancellationToken cancellationToken) => SendAsync(HttpMethod.Delete, requestUri, null, null, null, null, cancellationToken, IsSuccessStatusCodeFunc, false);
```

Comment:
delete needs query params too

Reply:
Probably but it wasn't required for this particular change. I can do it in this PR though.

Reply:
ok.  i need it for the resourceLocking.